### PR TITLE
Refactor stack recipes

### PR DESCRIPTION
**Pull request by tigercat2000**
- Styling fixes
 - Added post_build() to stack recipes, for any recipes that need to do
   something special
   - Called with two arguments, the stack that it originated from, and the
     new object.
 - Moved stack recipes to their own file

Currently, only two things use post_build
 - Cablecuffs use it to color the cuffs the same as the wire they
   originated from
 - Metal rods use it to call update_icon, in order to display the correct
   sprite for whatever amount of rods you created.

:cl:
rscadd: Readded colored cable-cuffs.
tweak: Making stacks of metal rods now updates the icon correctly.
/:cl:

### CODE REVIEW

File: code/game/objects/items/stacks/stack.dm [hunk 1]
```diff
@@ -10,156 +10,171 @@
...
 	user.set_machine(src) //for correct work of onclose
+
 	var/list/recipe_list = recipes
 	if(recipes_sublist && recipe_list[recipes_sublist] && istype(recipe_list[recipes_sublist], /datum/stack_recipe_list))
 		var/datum/stack_recipe_list/srl = recipe_list[recipes_sublist]
 		recipe_list = srl.recipes
+
 	var/t1 = "Amount Left: [amount]<br>"
-	for(var/i=1;i<=recipe_list.len,i++)
+	for(var/i = 1, i <= recipe_list.len, i++)
```
Comment (monster860) [inline code/game/objects/items/stacks/stack.dm]: Shouldn't this be

`for(var/i in 1 to recipe_list.len)` ?

File: code/game/objects/items/stacks/stack.dm [hunk 2]
```diff
@@ -10,156 +10,171 @@
...
+	if(usr.incapacitated() || usr.get_active_hand() != src)
+		return 0
 
 	if(href_list["sublist"] && !href_list["make"])
 		list_recipes(usr, text2num(href_list["sublist"]))
 
 	if(href_list["make"])
-		if(src.amount < 1) del(src) //Never should happen
+		if(amount < 1)
+			del(src) //Never should happen
```
Comment (monster860) [inline code/game/objects/items/stacks/stack.dm]: :eyes:

Why del()?

File: code/game/objects/items/stacks/stack.dm (hunk 1 above)
Comment (tigercat2000) [inline code/game/objects/items/stacks/stack.dm]: fuck that syntax

File: code/game/objects/items/stacks/stack.dm (hunk 2 above)
Comment (tigercat2000) [inline code/game/objects/items/stacks/stack.dm]: According to the comments all over the file, someone couldn't get stacks to GC inside a robot, so qdel broke everything.

**Comment by Lady-Luck** [comment]
Sorry tiger, all cable cuffs have to be red. There's no way to fix this.

File: code/game/objects/items/stacks/stack.dm (hunk 1 above)
Comment (monster860) [inline code/game/objects/items/stacks/stack.dm]: But that syntax looks better and is faster?

Comment (Fox-McCloud) [inline code/game/objects/items/stacks/stack.dm]: Use the syntax @monster860 mentioned; it's easier to read while also being faster.

File: code/game/objects/items/stacks/stack.dm (hunk 2 above)
Comment (Fox-McCloud) [inline code/game/objects/items/stacks/stack.dm]: This should still be qdel'd and stacks just return HARD_DEL for their qdel hint, then.

File: code/game/objects/items/stacks/stack.dm (hunk 1 above)
Comment (tigercat2000) [inline code/game/objects/items/stacks/stack.dm]: @Fox-McCloud I have literally no reason to take any advice from you on this, but fine, whatever.

File: code/game/objects/items/weapons/handcuffs.dm [hunk 3]
```diff
@@ -65,7 +65,7 @@
 /obj/item/weapon/restraints/handcuffs/cable
 	name = "cable restraints"
 	desc = "Looks like some cables tied together. Could be used to tie something up."
-	icon_state = "cuff_red"
+	icon_state = "cuff_white"
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/handcuffs.dm]: going to break the in-hand icon.

File: code/game/objects/items/stacks/stack.dm (hunk 2 above)
Comment (tigercat2000) [inline code/game/objects/items/stacks/stack.dm]: That's way beyond the scope of this PR.

Comment (Fox-McCloud) [inline code/game/objects/items/stacks/stack.dm]: It's beyond the scope of the PR to literally change a single `del` to `qdel(src)` and have. 

```
/obj/item/stack/Destroy()
	..()
	return QDEL_HINT_HARDDEL_NOW
```

Comment (tigercat2000) [inline code/game/objects/items/stacks/stack.dm]: This isn't the only del stacks use, the behaviour is spread over multiple files.

File: code/game/objects/items/weapons/handcuffs.dm (hunk 3 above)
Comment (tigercat2000) [inline code/game/objects/items/weapons/handcuffs.dm]: It works fine.

**Comment by Spacemanspark** [comment]
:+1:

File: code/game/objects/items/stacks/stack.dm (hunk 2 above)
Comment (Krausus) [inline code/game/objects/items/stacks/stack.dm]: Oh hey, I remember those dels. That was from back when our GC wasn't performing hard deletes whatsoever, and cyborg module code was built on the assumption that anything would be nulled out if it was deleted (probably still is), so they'd end up with broken phantom stacks stuck in their module list forever.

If cyborg module code was ever fixed to react to items not being inside of them anymore, those dels should all be qdels. Probably no point messing with them if cyborg code still doesn't handle items correctly, though.

File: code/game/objects/items/weapons/handcuffs.dm (hunk 3 above)
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/handcuffs.dm]: Huh...well then.

**Pull request merged**